### DISCUSSION TITLE
Simulating a Noise Sensor with a Contiki-NG IoT Device

### DISCUSSION THREAD

**Discussion by Tagliente-Simone** [contiki-ng/contiki-ng#1736] [Q&A] [answered]
Hello everyone!
Does anyone know how can I simulate readings of a sensor on a contiki-ng device? My idea was to create a file with the readings inside that will be parsed and then processed by the device but I actually don't know if it is a good idea or not and more if contiki-ng supports this kind of stuff. Thank you in advance, I'm available for better explanation of the problem.

**Comment by atiselsts** [comment, marked as answer]
Hi, you can look at these answers on how to simulate sensor readings in Cooja:
https://stackoverflow.com/questions/34881891/light-and-temperature-sht11-sensors-in-zolertia-z1/34884025#34884025
https://stackoverflow.com/questions/28013883/simulating-different-temperatures-on-cooja-contiki/28014418#28014418

If you want to simulate a sensor in Contiki-NG's C code, it's much simpler: just implement the sensor reading as a function that returns a random number from the desired distribution.

**Reply by Tagliente-Simone** [reply]
Thank you! It seems exactly what I was looking for. Cheers :)
